elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj

```diff
@@ -26,3 +28,62 @@
   "Performs a scroll query, fetching the next page of results from a query given a scroll id"
   [conn scroll-id opts]
   (doc/scroll conn scroll-id opts))
+
+(defn doc-get
+  "Fetches and returns a document by id or `nil` if it does not exist."
+  ([conn index mapping-type id]
+   (doc-get conn index mapping-type id nil))
+  ([conn index mapping-type id opts]
+   (let [result (if (empty? opts)
+                  (rest/get conn (rest/record-url conn index "_doc" id))
+                  (rest/get conn (rest/record-url conn index "_doc" id) {:query-params opts}))]
+     (if (not-found? result)
+       nil
+       result))))
+
+(defn put
+  "Creates or updates a document in the search index, using the provided document id"
+  ([conn index mapping-type id document]
+   (put conn index mapping-type id document nil))
+  ([conn index mapping-type id document opts]
+   (if (empty? opts)
```

Comment:
i dont really understand why we need this check and can't just pass the nil opts to `query-params` but if it works i won't complain

Reply:
I added this to solve a NullPointerException, but I don't think it is the cause of that. I will remove this check.